File: model/src/main/java/work/beltran/discogsbrowser/api/network/login/AccessHeader.java
package work.beltran.discogsbrowser.api.network.login;

/**
 * Created by Miquel Beltran on 14.05.16.
 * More on http://beltran.work
 */
public class AccessHeader extends LoginHeader {
    private String userToken;
    private String verifier;

    public AccessHeader(String consumerKey, String consumerSecret, String userToken, String verifier) {
        super(consumerKey, consumerSecret);
        this.userToken = userToken;
        this.verifier = verifier;
    }

    @Override
    public String getHeader() {
        return super.getHeader() + ", "
                + "oauth_verifier=\"" + verifier + "\", "
                + "oauth_token=\"" + userToken + "\"";
    }
}

File: business/src/main/java/work/beltran/discogsbrowser/business/login/AccessHeader.kt
package work.beltran.discogsbrowser.business.login

/**
 * Created by Miquel Beltran on 14.05.16.
 * More on http://beltran.work
 */
class AccessHeader(consumerKey: String,
                   consumerSecret: String,
                   userToken: String,
                   verifier: String)
    : LoginHeader(consumerKey, consumerSecret) {

    override val header = super.header + ", oauth_verifier=\"$verifier\", oauth_token=\"$userToken\""
}
